src/MAGMA.jl

```diff
@@ -1,5 +1,104 @@
 module MAGMA
 
-greet() = print("Hello World!")
+export device_reset,
+       MagmaVector, MagmaMatrix, get,
+       axpy!, dot,
+       gemv!
 
-end # module
+import Base: length,size,dot
+
+using  CUDArt
+import CUDArt: device_reset
+
+const magmatypes = Union{Float32, Float64, Complex64, Complex128}
+const magmatypeslist =  (Float32,Float64,Complex64,Complex128)
+const typechar = Dict(
+    Float32 => 's',
+    Float64 => 'd',
+    Complex64 => 'c',
+    Complex128 => 'z'
+)
+
+type MagmaVector{T <: magmatypes}
```

Comment:
This is julia v0.4 gramma. in v1.0+ we use `struct ... end`, check docs.julialang.org for details